### PR TITLE
Custom Model Classes: Ensure isDirty for embedded records doesn't recursively loop

### PR DESCRIPTION
**Pull request by igorT**
With Custom Model Class, we were notifying property changes for `isDirty`  on embedded models, which normally depend on their parents `isDirty` state, causing a loop.

### CODE REVIEW

File: config/ember-try.js [hunk 1]
```diff
@@ -48,8 +48,8 @@ module.exports = function () {
           name: 'ember-lts',
           npm: {
             devDependencies: {
-              'ember-source': '~3.16.0',
-              'ember-data': '~3.16.0',
+              'ember-source': '~3.24.0',
+              'ember-data': '~3.24.0',
```
Comment (rwjblue) [inline config/ember-try.js:52]: Let's do these changes in a dedicated PR, so we can have a nice entry in the changelog.

File: tests/unit/model/state-test.js [hunk 2]
```diff
@@ -1,8 +1,10 @@
 import { module, test, skip } from 'qunit';
-import { setupTest } from 'ember-qunit';
-
+import { setupTest, setupRenderingTest } from 'ember-qunit';
 import { run } from '@ember/runloop';
 import { isArray } from '@ember/array';
+import Component from '@ember/component';
+import hbs from 'htmlbars-inline-precompile';
```
Comment (rwjblue) [inline tests/unit/model/state-test.js:6]:
```suggestion
import { hbs } from 'ember-cli-htmlbars';
```

**Review by rwjblue — CHANGES_REQUESTED** [review]

File: config/ember-try.js (hunk 1 above)
Comment (igorT) [reply config/ember-try.js]: done

**Review by spham92 — APPROVED** [review]
lgtm

File: package.json [hunk 3]
```diff
@@ -64,6 +64,7 @@
     "broccoli-funnel": "^3.0.8",
     "ember-cli-babel": "^7.26.6",
     "ember-cli-version-checker": "^5.1.2",
+    "ember-compatibility-helpers": "^1.2.4",
```
Comment (rwjblue) [inline package.json:67]: devDep

**Review by rwjblue — APPROVED** [review]
One minor thing, but other than that LGTM

File: package.json (hunk 3 above)
Comment (igorT) [reply package.json]: done
